spec/features/team/invite_a_team_member_spec.rb

```diff
@@ -64,14 +66,55 @@
       end
 
       context "with an email that already exists" do
-        let(:invited_user_email) { user.email }
+        context "when the user is confirmed" do
+          let(:invited_user_email) { user.email }
+
+          it "tells the user that the email has already been taken" do
+            expect {
+              click_on "Send invitation email"
+            }.to change { User.count }.by(0)
+            expect(InviteUseCaseSpy.invite_count).to eq(0)
+            expect(page).to have_content("Email is already associated with an account. If you can't sign in, reset your password")
+          end
+        end
 
-        it "tells the user that the email has already been taken" do
-          expect {
-            click_on "Send invitation email"
-          }.to change { User.count }.by(0)
-          expect(InviteUseCaseSpy.invite_count).to eq(0)
-          expect(page).to have_content("Email is already associated with an account. If you can't sign in, reset your password")
+        context "when user is not confirmed and has no organisation" do
+          before do
+            sign_out
+            sign_up_for_account(email: invited_user_email)
+            sign_in_user user
+            visit root_path
+            click_on "Team"
+            click_on "Invite team member"
```

Comment:
This all seems very verbose:
```
visit root_path
click_on "Team"
click_on "Invite team member"
```
You can either visit the path directly or even manually generate the invitation?

Reply:
While I agree that it's verbose, I disagree with your proposed solutions.

The tests that matter are around inviting the team member, so these should remain.

I think the part that should be broken out to be programmatic should be signing up the new user.
```rb
User.create(email: invited_user_email, confirmed_at: nil)
```

Reply:
Given that "inviting team" is performed in the [outer context](https://github.com/alphagov/govwifi-admin/blob/c3d2cdb16ba0da5803f09ba35a1d9462c8e16b5f/spec/features/team/invite_a_team_member_spec.rb#L22-L25), you should be able to drop the 3 lines @tommotaylor mentioned, and just leave this `before` block for signing up a new user